### PR TITLE
feat: /availability - date overrides: Allow user to select multiple days 

### LINKED ISSUE #7909: [CAL-1324] /availability - date overrides: Allow user to select multiple days
[https://github.com/calcom/cal.com/assets/8019099/cc3ea8ba-2a3d-4c90-9b17-1d3256cf8ed4](https://github.com/calcom/cal.com/assets/8019099/cc3ea8ba-2a3d-4c90-9b17-1d3256cf8ed4)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1324](https://linear.app/calcom/issue/CAL-1324/availability-date-overrides-allow-user-to-select-multiple-days)</sub>

### PR DESCRIPTION
**Pull request by mehuylsharma**
## What does this PR do?

Adds the ability to select multiple days on adding overrides.

This PR is currently WIP. 

Fixes #7909 (issue)

starter

**Environment**: Staging(main branch) / Production

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

- [ ] Test A
- [ ] Test B

## Checklist

- I haven't performed a self-review of my own code and corrected any misspellings
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

/claim #7909

### CODE REVIEW

**Comment by mehuylsharma** [comment]
Tasks left: 

- Add the actual overrides 😅 
- Do not include `excludedDates` in override.

**Comment by mehuylsharma** [comment]
Hey @zomars, I have a few questions regarding this PR that I'd love to get the answers to, but I'm unsure if bounties v/s contributions are treated differently and bounties are "on-your-own" as in I need to figure it out on my own without any assistance. 🤔 

Anyway, currently this PR doesn't add the overrides to the `Date Overrides` box:

image

As far as I know, in `packages/features/schedules/components/DateOverrideInputDialog.tsx`:

```ts
const form = useForm({
    values: {
      range: value
        ? value.map((range) => ({
            start: new Date(
              dayjs
                .utc()
                .hour(range.start.getUTCHours())
                .minute(range.start.getUTCMinutes())
                .second(0)
                .format()
            ),
            end: new Date(
              dayjs.utc().hour(range.end.getUTCHours()).minute(range.end.getUTCMinutes()).second(0).format()
            ),
          }))
        : (workingHours || []).reduce((dayRanges, workingHour) => {
            if (date && workingHour.days.includes(date.day())) {
              dayRanges.push({
                start: dayjs.utc().startOf("day").add(workingHour.startTime, "minute").toDate(),
                end: dayjs.utc().startOf("day").add(workingHour.endTime, "minute").toDate(),
              });
            }
            return dayRanges;
          }, [] as TimeRange[]),
    },
  });
  ```
  
since `values` is an object and not an array of objects, it means that the Date Overrides box displays it as:

image

and not as separate dates (all the 9.00AM - 5.00PM ranges are 5 different dates).

Does this mean to refactor the `range` object and make it an array of objects (and display it accordingly in the UI)? Or am I possibly doing something wrong here?

Sorry for the long text wall 😅

**Comment by PeerRich** [comment]
this is cool @mehuylsharma whats missing to merge this?

**Comment by mehuylsharma** [comment]
> this is cool @mehuylsharma whats missing to merge this?

Sorry I'm taking so long currently stuck with something else, but I'll do an update in the next 24-48h hopefully with all of it working! Apologies for the delay 🙇 

As I said in [here](https://github.com/calcom/cal.com/pull/8954#issuecomment-1555901253) there might be a bit of refactoring with the current `ranges` object as it's not displaying different dates as different dates rather multiple time range, anyway I'll do my best version with it and reply back soon! 

Thanks! :D

File: apps/web/pages/availability/[schedule].tsx [hunk 1]
```diff
@@ -56,6 +56,18 @@ const DateOverride = ({ workingHours }: { workingHours: WorkingHours[] }) => {
   const { remove, append, update, fields } = useFieldArray<AvailabilityFormValues, "dateOverrides">({
     name: "dateOverrides",
   });
+
```
Comment (mehuylsharma) [inline apps/web/pages/availability/[schedule].tsx:59]: @sean-brydon I've implemented this to display all the dates separately in the UI, but the issue is the key of all these elements is the same so the delete/edit button does not function properly.

What could be the solution here? 🤔 

Thanks for the help again! 🙇

**Pull request closed without merging**